### PR TITLE
chore: Router refactor, bring back lost tests file, fix namings

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
A follow-up PR addressing the remaining comments from 
https://github.com/rakutentech/ios-inappmessaging/pull/26

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: RInAppMessaging/Classes/Extensions/UIView+IAM.swift [hunk 1]
```diff
@@ -20,6 +20,18 @@ extension UIView {
...
     }
+
+    func findIAMViewSubview() -> UIView? {
+        for subview in subviews {
+            if subview is BaseView {
+                return subview
+            } else if let iamView = subview.findIAMViewSubview() {
+                return iamView
+            }
+        }
```
Comment (malayli) [inline RInAppMessaging/Classes/Extensions/UIView+IAM.swift:31]: I'm wondering if you can use subviews.first{}:
```
let result = subviews.first {
    if $0 is BaseView {
        return true
    } else if let iamView = $0.findIAMViewSubview() {
        return true
    }
}
return result
```

Comment (imaginaris) [reply RInAppMessaging/Classes/Extensions/UIView+IAM.swift:31]: Hmm, it's going to break from the loop on the first element that returns true... so yeah I think it should work 👍

Comment (imaginaris) [reply RInAppMessaging/Classes/Extensions/UIView+IAM.swift:31]: Ah not really, the nested `$0.findIAMViewSubview()` call returns UIView which I have to discard. Returning true in that case will end up `result` being a parent view

**Review by donnie-jp — APPROVED** [review]
LGTM

**Review by malayli — APPROVED** [review]

**Pull request merged**